test/inductor/test_torchinductor.py

```diff
@@ -243,6 +244,52 @@ def run(*ex, **kwargs):
                 if exact_dtype:
                     assert correct_val.dtype == actual_val.dtype
 
+    if check_gradient:
+
+        def gather_leaf_tensors(args, kwargs):
+            leaf_tensors = []
+            args, args_spec = tree_flatten(args)
+            kwargs, kwargs_spec = tree_flatten(kwargs)
+            args = args + kwargs
+            for arg in args:
+                if not isinstance(arg, torch.Tensor):
+                    continue
+                if arg.requires_grad:
+                    leaf_tensors.append(arg)
+            return leaf_tensors
```

Comment:
spec is unused, and args and kwargs are both flattened - can we do something like:

`leaf_tensors = [arg for arg in tree_flatten(args, kwargs)[0] if isinstance(arg, torch.Tensor)]`

(Note: did not run the above, but list comprehension should be able to one line it)

Reply:
update.